### PR TITLE
Refactor box longhands

### PR DESCRIPTION
**Pull request by canova**
Converted some longhands into `vector_longhand` to reduce some repeated codes.
Also some of longhands with same `SpecifiedValue` and `computed_value::T` changed. Re-exported SpecifiedValue instead of computed_value::T. Normally a computed_value can't have a parse function.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

- [X] These changes do not require tests because they're not changing the behavior of the longhands.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14913)

### CODE REVIEW

File: tests/unit/style/properties/serialization.rs [hunk 1]
```diff
@@ -536,10 +536,11 @@ mod shorthand_serialization {
     fn transition_should_serialize_all_available_properties() {
         use euclid::point::Point2D;
         use style::properties::animated_properties::TransitionProperty;
+        use style::properties::longhands::transition_delay::SpecifiedValue as DelayContainer;
         use style::properties::longhands::transition_duration::SpecifiedValue as DurationContainer;
         use style::properties::longhands::transition_property::SpecifiedValue as PropertyContainer;
         use style::properties::longhands::transition_timing_function::SpecifiedValue as TimingContainer;
-        use style::properties::longhands::transition_timing_function::TransitionTimingFunction;
+        use style::properties::longhands::transition_timing_function::single_value::SpecifiedValue as TransitionTimingFunction;
```
Comment (canaltinova) [inline tests/unit/style/properties/serialization.rs]: How can I indent this properly?

**Comment by Manishearth** [comment]
I think empty vectors were allowed for many of these properties. You need to explicitly enable that with `allow_empty=True`. Aside from that, looks good to me

@bors-servo delegate+

File: tests/unit/style/properties/serialization.rs (hunk 1 above)
Comment (Manishearth) [inline tests/unit/style/properties/serialization.rs]: I don't think you can. `use style::properties::longhands::transition_timing_function` and then `transition_timing_function::whatever` in the code might work better (replace all TTF imports if you're doing this)

**Comment by canaltinova** [comment]
I especially looked for `allow_empty` but it looks like other properties don't allow `none` keyword. Generally other properties are like this: `<time> #`

**Comment by jdm** [comment]
@bors-servo: retry
* #14323

**Pull request merged**